ENIGMAsystem/SHELL/Collision_Systems/Polygon/placeholderlinks.cpp

```diff
@@ -0,0 +1,166 @@
+/** Copyright (C) 2021 Nabeel Danish 
+***
+*** This file is a part of the ENIGMA Development Environment.
+***
+*** ENIGMA is free software: you can redistribute it and/or modify it under the
+*** terms of the GNU General Public License as published by the Free Software
+*** Foundation, version 3 of the license or any later version.
+***
+*** This application and its source code is distributed AS-IS, WITHOUT ANY
+*** WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+*** FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+*** details.
+***
+*** You should have received a copy of the GNU General Public License along
+*** with this code. If not, see <http://www.gnu.org/licenses/>
+**/
+
+#include "Collision_Systems/collision_mandatory.h"
+#include "Universal_System/nlpo2.h"
+#include "Universal_System/Resources/sprites_internal.h"
+
+#include <iostream>
+
+// Cross product.
+static inline double cp(double x1, double y1, double x2, double y2) { return x1*y2 - x2*y1; }
+static inline int min(int x, int y) { return x<y? x : y; }
+static inline int max(int x, int y) { return x>y? x : y; }
+
+namespace enigma
+{
+  // A non-NULL pointer is a sprite mask, a NULL pointer means bbox should be used.
+  void *get_collision_mask(const Sprite& spr, void* input_data, collision_type ct) // It is called for every subimage of every sprite loaded.
+  {
+    
+    unsigned char* data = (unsigned char*)input_data;
+    switch (ct)
+    {
+      case ct_precise:
+        {
+          const unsigned int w = spr.width, h = spr.height;
+          unsigned char* colldata = new unsigned char[w*h];
+
+          for (unsigned int rowindex = 0; rowindex < h; rowindex++)
+          {
+            for(unsigned int colindex = 0; colindex < w; colindex++)
+            {
+              colldata[rowindex*w + colindex] = (data[4*(rowindex*w + colindex) + 3] != 0) ? 1 : 0; // If alpha != 0 then 1 else 0.
+            }
+          }
+
+          return colldata;
+        }
+      case ct_bbox: return 0;
+      case ct_ellipse:
```

Comment:
Did we ever discuss why you needed to generate bitmasks for primitive shapes?

Reply:
No, I don't think so. This is a file that I do not fully understand what is doing, but I included it because it is found in other collision systems and I thought that I will implement it as well.